modules/programs/navi.nix

```diff
@@ -0,0 +1,85 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+
+  cfg = config.programs.navi;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.navi = {
+    enable = mkEnableOption "Navi";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.navi;
+      defaultText = literalExpression "pkgs.navi";
+      description = "The package to use for the navi binary.";
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      defaultText = literalExpression "{ }";
+      example = literalExpression ''
+        {
+          cheats = {
+            paths = [
+              "~/cheats/"
+            ];
+          };
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/navi/config.yaml</filename> on Linux
+        or <filename>~/Library/Application Support/navi/config.yaml</filename> on Darwin. See
+        <link xlink:href="https://github.com/denisidoro/navi/blob/master/docs/config_file.md"/>
+        for more information.
+      '';
+    };
+
+    enableBashIntegration = mkEnableOption "Bash integration" // {
+      default = true;
+    };
+
+    enableZshIntegration = mkEnableOption "Zsh integration" // {
+      default = true;
+    };
+
+    enableFishIntegration = mkEnableOption "Fish integration" // {
+      default = true;
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [{
```

Comment:
Small nitpick, you don't have to change this but I don't think `mkMerge` is necessary here as there aren't multiple config values which need to be merged.

In addition, it might be worthwhile to abstract the config directory and set only `home.file` instead of having two mutually exclusive option declaration to avoid duplicate comparisons. This works because the fileType type will correctly resolve an absolute path and return it unmodified.

https://github.com/nix-community/home-manager/blob/829e89a16f4f96428d1b94e68d4c06107b5491c0/modules/lib/file-type.nix#L22

For example:
```nix
let
  configDir = if pkgs.stdenv.isDarwin then
    "Library/Application Support"
  else
    config.xdg.configHome;
in {
  config = mkIf cfg.enable {
    home.file."${configDir}/navi/config.yaml" =
      mkIf (cfg.settings != { }) {
        source = yamlFormat.generate "navi-config" cfg.settings;
      };
  };
}
```

Reply:
Agreed, I believe these changes are worth making :+1:

Reply:
Thank you for reviewing! I've addressed your comments PTAL